### ISSUE TITLE
Limited performance for user with no profile 

### ISSUE THREAD

**Issue by alexseyk** [derat/duoleavemealone#8]
Since there is no account / profile i can not buy "timed practice" >> start "non timed practice page" is not skipped. The button is actually called just "start practice" . Script log says "found next button".

It IS  the same  behavior for anyone without timed practice purchase.

At the end of the practice "Time to create a profile!" page is not bypassed :(
There is no next button , just small link "Create a profile later"

Does not work for placement test. & for users with profile too 

Please fix it at your leisure time, thanks

**Comment by derat** [comment]
Hmm, I didn't even know it was possible to use Duolingo without an account.

When auto-starting practice, the extension looks for a button with a `data-test` attribute of `secondary-button` and its `background-color` style set to `rgb(24, 153, 214)`. If there's just one button at the start of the practice page, it won't click it. I think it'll be hard to distinguish between the case where there's a single start button and when the user is at the first challenge.

It's probably easiest to just create a profile, if that's an option here. :-P

**Comment by alexseyk** [comment]
>between the case where there's a single start button and when the user is at the first challenge.

I see it's the same button data-test="player-next" and the only difference is button text "start practice" or  "start timed practice" or "check"

Creating a profile maybe "easier" and users without  a profile probably don't use scripts / extensions yet. It is just that  I like to test my language retention on a placement test.  For me it is easier to open Duo in an another (or incognito) browser than to reset a language on a secondary account. It is less mouse clicks anyway 

As you suggested 

    const isPractice = window.location.href.indexOf('/practice') != -1 ||
                       window.location.href.indexOf('/placement/') != 1;

Have done the trick :) 

PS:  Your script actually  works for users without profile, but ONLY if i don't CHANGE  account settings (i.e. turn microphone & sound effects off ).  or until i do first lesson.  Re-openning browser somehow helps.

**Comment by derat** [comment]
Thanks for trying it. I've added basic support for placement tests in 36f1fa36c44a376ad19776e93f15212ef237cc1b and have submitted 0.15 to the Web Store. If you're using an incognito window, there's a setting available under `chrome://extensions` for making the extension work there.

When trying a placement test today, I noticed that the "continue" button flickers while my mouse is hovering over it after I click it. It looks like it also happens with the extension disabled, though, so it's probably Duolingo's bug and not mine.
